FIB 1.3 Documentation Update (+Release Notes)

yamkovoy opened the pull request:
## Issues to keep track of:
- Inconsistent use of the "credits" and "tokens" terms.
  > Will be addressed in the coming days, after a possible discussion w/ @Mingela.\
  > Feel free to express your ideas in the comments.

Mingela commented on src/api-specification/contribution-controller/assembling-a-contribution.md:
```diff
@@ -8,24 +10,56 @@
-    origination: 'string', //two-digit country code (Alpha-2, ISO 3166)
-    destination: 'string', //two-digit country code (Alpha-2, ISO 3166)
-    expiryDate: 0 //integer($int32), Unix Epoch Time in seconds
+    "id": "idRange", //127.0.0.1 OR 127.0.0.1-127.0.0.2 OR +14155552671 OR +14155552671-+14155552672 OR 107615702016566
```
I'd remove examples from the comment section and place them in the contribution format description page

Mingela commented on src/api-specification/contribution-controller/retrieving-a-contribution-by-ID.md:
```diff
@@ -0,0 +1,102 @@
+    {
+        "assetDefinitionIds": "idRange#contribution",
+        "contribution": {
+            "id": "idRange", //127.0.0.1 OR 127.0.0.1-127.0.0.2 OR +14155552671 OR +14155552671-+14155552672 OR 107615702016566
```
from here and potentially other places, applicable to any attribute, not just id

Mingela commented on src/api-specification/contribution-controller/retrieving-a-contribution-by-ID.md:
```diff
@@ -0,0 +1,102 @@
+            "origination": "string",
+            "destination": "string",
+            "expiryDate": integer($int32),
+            "fraudStatus": "string(enum)", //'ACTIVE' OR 'EXPIRED' OR 'FLAGGED'
```
and this etc

Mingela commented on src/api-specification/contribution-controller/retrieving-contributions.md:
```diff
@@ -0,0 +1,136 @@
+| Field              | Value Type        | Description                                                                                                                                                                                                                                                                                                                                                          ...
+| :----------------: | ----------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ...
+| `size`             | `integer($int32)` | The number of entries, starting with the latest, shown in the response. <br> Set to `50` by default.                                                                                                                                                                                                                                                                 ...
+| `from` and `to`    | `string`          | The timeframe that the response entries are filtered by (set as [Unix Epoch time](https://www.epochconverter.com/clock) in seconds). <br> The `from` parameter is set to `0` by default.                                                                                                                                                                             ...
```
I find this merging of fields questionable

Mingela commented on src/api-specification/contribution-controller/retrieving-contributions.md:
```diff
@@ -0,0 +1,136 @@
+| `org`              | `array[string]`   | The country of origination of the fraud event. Set as a two-letter country code (Alpha-2, ISO 3166; e.g., US, GA).                                                                                                                                                                                                                                                   ...
+| `self-only`        | `boolean`         | A boolean that defines whether the response should only contain contributions submitted by the requesting peer. <br> The `self-only` parameter is set to `false` by default.                                                                                                                                                                                         ...
+| `fetch-mode`       | `string`          | The parameter that filters the response entries based on whether the requesting user has already seen them. <br> Can be one of the following: <ol><li>`NEW` — includes only the unseen by the requesting user entries.</li><li>`OLD` — includes only the already seen by the requesting user entries.</li><li>`DEFAULT` — includes all of the entries.</li></ol> <br ...
+| `confidence-score` | `boolean`         | A boolean that defines whether the price of the retrieved contributions is affected by the [confidence index](../../overview/tokenomics.md#confidence-index). If set to `true`, the price of any given contribution is expected to rise above the [default conversion rate](../../overview/tokenomics.md#current-conversion-rate). <br> The `confidence-score` param ...
```
```suggestion
| `confidence-score` | `boolean`         | A boolean that defines whether the price of the retrieved contributions is affected by the [confidence index](../../overview/tokenomics.md#confidence-index). If set to `true`, the price of any given contribution might rise above the [default conversion rate](../../overview/tokenomics.md#current-conversion-rate). <br> The `confidence-score` parameter is set to `false` by default.                   |
```

Actually we could just highlight the way of getting relevant prices instead

Mingela commented on src/api-specification/contribution-controller/retrieving-contributions.md:
```diff
@@ -0,0 +1,136 @@
+          "timestamp": integer($int32)
+        }
+    ],
+    "details": {
```
I guess it deserves more elaboration (refer to Figma states of this operation).
There can be:
* Full retrieval (enough credits)
* Partial retrieval (not enough credits)
* Empty retrieval (not enough credits)
* Empty retrieval (no data for the filters specified)

Mingela commented on src/api-specification/contribution-controller/retrieving-rewards-rate.md:
```diff
@@ -0,0 +1,120 @@
+# Retrieving rewards rate
+
+> Retrieves the current rewards rate established for [submitting contributions](submitting-a-contribution.md) of all fraud types for all registered company entities.
```
```suggestion
> Retrieves the current rewards rate established for [submitting contributions](submitting-a-contribution.md) of all fraud types for all supported company types.
```

Mingela replied:
And flagging?

Mingela commented on src/api-specification/contribution-controller/submitting-a-contribution-flag.md:
```diff
@@ -0,0 +1,81 @@
+# Submitting a contribution flag
+
+> Submits [an assembled contribution flag](assembling-a-contribution-flag.md).
```
should there be a reference to the general section with overview what the flag is?

Mingela commented on src/api-specification/peer-controller/retrieving-peer-limit-status.md:
```diff
@@ -1,5 +1,7 @@
 # Retrieving peer limit status
 
+> Retrieves the limit status of the [authorized FIB user](../auth-controller/authorizing-a-user-in-the-system.md) at the time of the request.
```
status is kinda 'exceeded/not' value to me, how about reformatting that to `state` instead?

Mingela commented on src/api-specification/peer-controller/retrieving-peer-limit-status.md:
```diff
@@ -34,12 +34,50 @@ A `GET` request to the endpoint.
+    "status": {
+        "code": 200,
+        "name": "Ok",
+        "message": "Peer limit status has been retrieved successfully"
```
there will be an empty message if success (for any request)

Mingela commented on src/api-specification/peer-controller/retrieving-user-details.md:
```diff
@@ -0,0 +1,99 @@
+    "isPrivileged": false,
+    "registrationDate": "2024-01-20T10:02:50.118297",
+    "subscriptionType": "FREEMIUM",
+    "subscriptionExpiryDate": "2024-12-03T10:15:30.118297"
```
there can't be expiration for FREEMIUM susbcription, either set that to `PAID` or expiry to `null`

Mingela commented on src/api-specification/wallet-controller/retrieving-token-balance.md:
```diff
@@ -0,0 +1,69 @@
+# Retrieving token balance
+
+> Retrieves the token balance on the [authorized FIB user's](../auth-controller/authorizing-a-user-in-the-system.md) account at the time of the request.
```
```suggestion
> Retrieves the token balance on the [authorized FIB user's](../auth-controller/authorizing-a-user-in-the-system.md) peer at the time of the request.
```

Mingela commented on src/api-specification/wallet-controller/retrieving-token-balance.md:
```diff
@@ -0,0 +1,69 @@
+| Response Code | Description                                    |
+| :-----------: | ---------------------------------------------- |
+| `200`         | Token balance has been retrieved successfully. |
+| `401`         | ???                                            |
```
401 is always Unauthorized, when authentication token invalid/absent or expired

DrLightman13 replied:
403 may return as well - Forbidden (in case a non-admin peer attempts to request the credit balance of another peer)

Mingela commented on src/getting-started.md:
```diff
@@ -1,10 +1,16 @@
 # Getting started
 
-This section is designed to help you integrate the FIB API seamlessly into your application or system by providing instructions on how to register an account, check the API availability, call API endpoints, and test your implementation in a sandbox environment.
+Since FIB allows its users to interact with the network both via API and the Web App, this section provides instructions for all necessary preliminary steps, be you a regular user or a developer aiming to integrate the FIB API into their application or system.
```
```suggestion
Since FIB allows its users to interact with the network both via API and the Web App, this section provides instructions for all necessary preliminary steps, be you a regular user or a developer aiming to integrate the FIB API into their application or system.
```
```suggestion
Since FIB allows its users to interact with the network both via API and the Web App, this section provides instructions for all necessary preliminary steps, be you a frontend application user or a developer aiming to integrate the FIB API into their application or system.
```

Mingela commented on src/getting-started.md:
```diff
@@ -1,10 +1,16 @@
+Since FIB allows its users to interact with the network both via API and the Web App, this section provides instructions for all necessary preliminary steps, be you a regular user or a developer aiming to integrate the FIB API into their application or system.
 
-It is advised to read the following topics in the order that they are presented:
+Regular users should start with the [Registering an account](getting-started/registering-an-account.md) tutorial, and then move on to the [Tutorials: Web App](tutorials-web.md) section that explains how to interact with the FIB network via [FIB Web App](overview/web-interface.md).
```
```suggestion
Users should start with the [Registering an account](getting-started/registering-an-account.md) tutorial, and then move on to the [Tutorials: Web App](tutorials-web.md) section that explains how to interact with the FIB network via [FIB Web App](overview/web-interface.md).
```

For anyone it's crucial to understand the flows imo

Mingela replied:
Moreover registering and navigating to the profile page is required even for solely API integrator

Mingela commented on src/getting-started/registering-an-account.md:
```diff
@@ -1,30 +1,91 @@
-3. Agree to the [Terms & Conditions](https://github.com/fraud-intelligence-limited/fil-legal/blob/main/Fraud%20Intelligence%20Blockchain%20Terms%20of%20Use%20v4.0.pdf) by selecting the **I agree to the terms** checkbox.
-4. On the **Subscription details** tab, select **Register Account**.
+1. Go to the [Fraud Intelligence Blockchain website](https://app.fraudintelligencelimited.com) and select **Sign up**.
+2. On the **Choose your plan** screen that appears, select the following:
```
this screen got removed, please adjust

Mingela commented on src/getting-started/registering-an-account.md:
```diff
@@ -1,30 +1,91 @@
+
+   :::
+
+7. Review the [Terms & Conditions](https://github.com/fraud-intelligence-limited/fil-legal/blob/main/Fraud%20Intelligence%20Blockchain%20Terms%20of%20Use%20v6.0.md), then select the corresponding **I agree with the Terms and Conditions** checkbox.
```
the version of that might be updated, let's better refer to the main branch of the repo only

Mingela commented on src/overview/tokenomics.md:
```diff
@@ -4,33 +4,41 @@ This topic provides a summary of the key financial concepts that are operational
+## Current conversion rate
+
+- **Uploading fraud events**: `10` tokens per `id` fraud identifier.
+- **Downloading fraud events**: `1` tokens per `id` fraud identifier.
```
```suggestion
- **Downloading fraud events**: `1` tokens per `id` fraud identifier.
```
```suggestion
- **Downloading fraud events**: `1` token per `id` fraud identifier.
```

Mingela commented on src/overview/tokenomics.md:
```diff
@@ -40,38 +48,38 @@ When assigning a contribution with a confidence index value, FIB considers many
 [//]: <> (If necessary, a detailed description of how the confidence index is calculated may be added here or to the "Contributions" topic)
 
-The confidence index value inherently affects the price of downloading the fraud event data. By default, peers download fraud event data in accordance with the [default conversion rate](#current-default-conversion-rate). However, when [retrieving contributions](../api-specification/contribution-controller/retrieving-top-contributions.md) (see the `confidence-score` parameter) a peer can specify w ...
+The confidence index value inherently affects the price of downloading the fraud event data. By default, users download fraud event data in accordance with the [current conversion rate](#current-conversion-rate). However, when [retrieving contributions](../api-specification/contribution-controller/retrieving-contributions.md) (see the `confidence-score` parameter) a user can specify whether they  ...
```
```suggestion
The confidence index value inherently affects the price of downloading the fraud event data. By default, users download fraud event data in accordance with the [current conversion rate](#current-conversion-rate). However, when [retrieving contributions](../api-specification/contribution-controller/retrieving-contributions.md) (see the `confidence-score` parameter) a user can specify whether they want the price of the retrieved contributions to be calculated in accordance with the confidence index. If enabled, the price of any given contribution is expected to rise above the [current conversion rate](#current-conversion-rate).
```
```suggestion
The confidence index value inherently affects the price of downloading the fraud event data. By default, users download fraud event data in accordance with the [current conversion rate](#current-conversion-rate). However, when [retrieving contributions](../api-specification/contribution-controller/retrieving-contributions.md) (see the `confidence-score` parameter) a user can specify whether they want the price of the retrieved contributions to be calculated in accordance with the confidence index. If enabled, the price of any given contribution might rise above the [current conversion rate](#current-conversion-rate).
```

Mingela reviewed (commented):
please also consider providing Vercel preview ready asap for visual review

DrLightman13 commented on src/api-specification/auth-controller/authorizing-a-user-in-the-system.md:
```diff
@@ -8,53 +10,86 @@
+| `200`         | User authorization has been successful.                                                         |
+| `401`         | User signature is incorrect or failed to be decoded.                                            |
+| `404`         | User has not been found.                                                                        |
+| `422`         | User signature either could not be decoded or verified, or the `publicKey` string is not valid. |
```
I think "publicKey" must be changed to "authPublicKeyHex".

DrLightman13 commented on src/api-specification/auth-controller/authorizing-a-user-in-the-system.md:
```diff
@@ -8,53 +10,86 @@
+| Response Code | Description                                                                                     |
+| :-----------: | ----------------------------------------------------------------------------------------------- |
+| `200`         | User authorization has been successful.                                                         |
+| `401`         | User signature is incorrect or failed to be decoded.                                            |
```
The request doesn't return a 401 error code, but it does return a 400 error code instead.

DrLightman13 commented on src/api-specification/auth-controller/logging-a-user-out-of-the-system.md:
```diff
@@ -0,0 +1,57 @@
+| Response Code | Description                                                                                                                                   |
+| :-----------: | --------------------------------------------------------------------------------------------------------------------------------------------- |
+| `200`         | User has been successfully logged out.                                                                                                        |
+| `401`         | Either user signature or `refreshToken` are not valid, or the `accessToken` bearer token could not be parsed from the `Authorization` header. |
```
The request returns a 400 error code as well.

DrLightman13 commented on src/api-specification/auth-controller/refreshing-authentication-tokens.md:
```diff
@@ -39,9 +56,9 @@ A `PATCH` request to the endpoint, where the `Authorization` header is required.
-| `500` | Internal server error. |
+| Response Code | Description                                                                                                                   |
+| :-----------: | ----------------------------------------------------------------------------------------------------------------------------- |
+| `200`         | Authentication tokens have been refreshed successfully.                                                                       |
```
The method may also return the following error codes: 

- 400; 
- 422.

DrLightman13 commented on src/api-specification/contribution-controller/assembling-a-contribution-flag.md:
```diff
@@ -8,47 +10,71 @@
+| Response Code | Description                                                                                           |
+| :-----------: | ----------------------------------------------------------------------------------------------------- |
+| `200`         | Contribution flag has been assembled and its `transactionHex` string has been retrieved successfully. |
+| `400`         | The request body of the contribution flag is incorrect or premium contribution can't be changed.      |
```
The method may also result in a 401 - Unauthorized.

DrLightman13 commented on src/api-specification/contribution-controller/assembling-a-contribution.md:
```diff
@@ -34,28 +68,45 @@ This endpoint accepts requests with up to `500` entries.
+| Response Code | Description                                                                                      |
+| :-----------: | ------------------------------------------------------------------------------------------------ |
+| `200`         | Contribution has been assembled and its `transactionHex` string has been retrieved successfully. |
+| `400`         | Contribution type is invalid.                                                                    |
```
401 may be returned as well.

DrLightman13 commented on src/api-specification/contribution-controller/retrieving-a-contribution-by-ID.md:
```diff
@@ -0,0 +1,102 @@
+
+### Responses
+
+| Response Code | Description                                                                   |
```
401 and 404 may be returned as well.

DrLightman13 commented on src/api-specification/contribution-controller/retrieving-contributions.md:
```diff
@@ -0,0 +1,136 @@
+      "self": integer($int32), //The number of contributions that have been submitted by the requesting user
+      "old": integer($int32), //The number of contributions that have already been seen by the requesting user
+      "new": integer($int32), //The number of contributions that have _not_ yet been seen by the requesting user
+      "tokensSpent": integer($int32), //The total amount of tokens spent on the retrieved contributions
```
As far as I know, it's "creditsSpent", not "tokensSpent".

Mingela replied:
@yamkovoy I'd refrain from any further changes to API considering risk it might bring, I agree we may implement unification though credits/tokens does not introduce confusion in the related scenario. Hope it would not irritate you much 😃

DrLightman13 commented on src/api-specification/contribution-controller/retrieving-contributions.md:
```diff
@@ -0,0 +1,136 @@
+| Response Code | Description                                                                              |
+| :-----------: | ---------------------------------------------------------------------------------------- |
+| `200`         | Contributions have been successfully retrieved and filtered by the specified parameters. |
+| `400`         | Request parameters are invalid or the fraud type could not be resolved by code.          |
```
+ 401 too

DrLightman13 commented on src/api-specification/contribution-controller/retrieving-pricing-rate.md:
```diff
@@ -0,0 +1,64 @@
+| :-----------: | ---------------------------------------------------- |
+| `200`         | The pricing rate has been successfully retrieved.    |
+| `401`         | User signature is incorrect or failed to be decoded. |
+| `404`         | Either account or its metadata key has not found.    |
```
404 is excessive here.

DrLightman13 commented on src/api-specification/contribution-controller/submitting-a-contribution-flag.md:
```diff
@@ -0,0 +1,81 @@
+| Response Code | Description                                                                                    |
+| :-----------: | ---------------------------------------------------------------------------------------------- |
+| `200`         | Contribution has been flagged and its `transactionHex` string has been retrieved successfully. |
+| `400`         | Invalid transaction.                                                                           |
```
It also returns 401 and 404.

DrLightman13 commented on src/api-specification/contribution-controller/submitting-a-contribution.md:
```diff
@@ -8,47 +10,84 @@
+| Response Code | Description                                                                  |
+| :-----------: | ---------------------------------------------------------------------------- |
+| 200           | Contribution has been submitted and tokens have been rewarded successfully. |
+| 400           | Invalid transaction.                                                         |
```
401 may be returned as well.

DrLightman13 commented on src/api-specification/peer-controller/retrieving-peer-limit-status.md:
```diff
@@ -34,12 +34,50 @@ A `GET` request to the endpoint.
+| Response Code | Description                                               |
+| :-----------: | --------------------------------------------------------- |
+| `200`         | Peer limit status has been retrieved successfully.        |
+| `404`         | Metadata key not found or Trigger's period not specified. |
```
401 returns, too.

DrLightman13 replied:
I am not sure if 404 is possible. It's better to clarify this one.

DrLightman13 commented on src/api-specification/peer-controller/retrieving-user-details.md:
```diff
@@ -0,0 +1,99 @@
+
+**Encoding**: `JSON`
+
+**Endpoint**: `/data/api/v1/peer-management/account`
```
```suggestion
**Endpoint**: `/data/api/v1/peer-management/details
```

DrLightman13 replied:
I believe the endpoint is wrong here. It should be "auth" instead of "data": auth/api/v1/peer-management/account.

DrLightman13 commented on src/api-specification/contribution-controller/assembling-a-contribution.md:
```diff
@@ -34,28 +68,45 @@ This endpoint accepts requests with up to `500` entries.
+| :-----------: | ------------------------------------------------------------------------------------------------ |
+| `200`         | Contribution has been assembled and its `transactionHex` string has been retrieved successfully. |
+| `400`         | Contribution type is invalid.                                                                    |
+| `404`         | Account not found or the rewards table doesn't exist.                                            |
```
Not possible.

DrLightman13 commented on src/tutorials-api/retrieving-contributions.md:
```diff
@@ -0,0 +1,141 @@
+      "self": integer($int32), //The number of contributions that have been submitted by the requesting user
+      "old": integer($int32), //The number of contributions that have already been seen by the requesting user
+      "new": integer($int32), //The number of contributions that have _not_ yet been seen by the requesting user
+      "tokensSpent": integer($int32), //The total amount of tokens spent on the retrieved contributions
```
```suggestion
      "creditsSpent": integer($int32), //The total amount of tokens spent on the retrieved contributions
```

DrLightman13 commented on src/tutorials-api/submitting-a-contribution/verifying-a-contribution-has-been-submitted.md:
```diff
@@ -1,93 +1,144 @@
+         "self": integer($int32), //The number of contributions that have been submitted by the requesting user
+         "old": integer($int32), //The number of contributions that have already been seen by the requesting user
+         "new": integer($int32), //The number of contributions that have _not_ yet been seen by the requesting user
+         "tokensSpent": integer($int32), //The total amount of tokens spent on the retrieved contributions
```
```suggestion
         "creditsSpent": integer($int32), //The total amount of tokens spent on the retrieved contributions
```

Mingela commented:
inside JSON examples string values are inside `".."`, not `'..'`

Mingela commented on src/api-specification/contribution-controller/assembling-a-contribution.md:
```diff
@@ -1,5 +1,7 @@
 # Assembling a contribution
 
+> Assembles a contribution that must be later [submitted](submitting-a-contribution.md) in order to store the fraud event data on the network.
```
first part is still relevant, `must be later` looks a bit strange to me

Mingela commented:
sometimes they're even combined, the first one is more natural, please make all consistent
![image](https://github.com/fraud-intelligence-limited/fil-documentation/assets/17858062/32d7e0bf-0362-48b7-a6c2-677abccef84b)

Mingela commented on .vitepress/config.mts:
```diff
@@ -40,7 +40,7 @@ export default defineConfig({
             },
             {
-              text: 'Web UI',
+              text: 'Web App UI',
```
please check all links once again, I can see `Tutorials: API -> Retrieving top contributions -> 404`, moreover let's get rid of `top`

yamkovoy commented:
### Changes:
- Fixed formatting of the code blocks to the one that the docs build and the Prettier script accepts as proper JSON5 formatting.
  > Unfortunately, it is not the one requested by @Mingela, but I couldn't make the build accept the docs otherwise.\
  > It is the same formatting as suggested by [https://spec.json5.org/](https://spec.json5.org/), which, I believe, the script follows.
- Had to break the formatting of callouts that were intended to be within spoliers (`::: details` callouts), now they're outside, which is not perfect, but should be acceptable.
- Checked for dead links, fixed.

Mingela commented:
Looks good, I'd like the following issues to be addressed as well:
* https://github.com/fraud-intelligence-limited/fil-documentation/issues/66
* https://github.com/fraud-intelligence-limited/fil-documentation/issues/37
* https://github.com/fraud-intelligence-limited/fil-documentation/issues/36

Mingela commented on src/release-notes/index.md:
```diff
@@ -6,20 +6,98 @@ next: false
+  > Documented here:\
+  > [Overview > Tokenomics](../overview/tokenomics.md#confidence-index) — Updated the existing **Note** callout.
+
+- Added support of dial-in codes for international networks (`+882` and `+883`).
```
For these, `origination` value should be left empty

Mingela commented:
please also consider comments from @DrLightman13, seems there are many relevant still

Mingela reviewed (dismissed):
do not merge, I'll do that myself by the release time
